docs/pages/reference/config.mdx

```diff
@@ -38,6 +38,63 @@ enable you to roll back to the previous configuration if you need to.
 
 </Notice>
 
+## Configuration file versions
+
+Teleport's configuration file is versioned independently from Teleport's
+releases. Teleport clusters currently support all versions of the configuration
+file. 
+
+The latest version is `v3`. If your configuration file has a lower version, we
+recommend moving to `v3` when possible for the best Teleport experience.
+
+<Notice type="danger">
+
+If a configuration file does not include a `version` field, the `teleport`
+binary reading the configuration assumes it is `v1`.
+
+</Notice>
+
+Here are the features that require users to opt in by explicitly upgrading their
+configuration to a newer version.
+
+### TLS Routing
+
+[TLS Routing](../management/operations/tls-routing.mdx) was introduced in
+Teleport 8 and configuration file version `v2`. With TLS Routing, Teleport's
+Proxy Service listens on a single port and uses ALPN and SNI to route incoming
+traffic to the correct listener rather than listening on multiple
+protocol-specific ports.
+
+For backwards compatibility, configuration version `v1` always disables TLS
```

Comment:
@smallinsky is this true?

Reply:
No, That seems to be not right. 

TLS Routing feature on the teleport proxy side is always enable - no matter if v1 or v2 configuration version was selected. 
So even in v1 config version the `proxy_listener_mode: multiplex` can be set  to tell teleport proxy clients to use TLS Routing to connect to Teleport proxy. 

The difference between v1 and v2 is the behavior  when the additional listeners addressed were not explicitly specified. https://github.com/gravitational/teleport/blob/master/rfd/0039-sni-alpn-teleport-proxy-routing.md#proxy-configuration

v1  config will start legacy listener by default even if the address like `tunnel_listen_addr` was not provided/was empty( the logic fallback to the default tunnel address).  
In v2 the behavior for empty proxy listener services is different. For instance if  the `tunnel_listen_addr` proxy setting was not provided the v2 config won't start the tunnel proxy on the default address. 

This v1 - v2 behavior applies to all the proxy listener expect webport where the TLS Routing logic is implemented.